src/pudl/settings.py

```diff
@@ -71,64 +71,94 @@ def validate_tables(cls, tables):  # noqa: N805
         return sorted(set(tables))
 
 
-def create_dataset_settings(name: str) -> GenericDatasetSettings:
+class Ferc1Settings(GenericDatasetSettings):
     """
-    Create a pydantic settings class for a dataset.
+    An immutable pydantic model to validate Ferc1Settings.
 
-    This function dynamically creates subclasses of GenericDatasetSettings
-    using information from the constants WORKING_PARTITIONS and PUDL_TABLES.
-    The returned class can be subclassed to add additional fields or validators.
+    Parameters:
+        years: List of years to validate.
+        tables: List of tables to validate.
+
+        working_partitions: Dictionary of working paritions.
+        working_tables: List of working tables.
+    """
+
+    working_partitions: ClassVar = DataSource.from_id("ferc1").working_partitions
+    working_tables: ClassVar = sorted(list(pc.PUDL_TABLES["ferc1"]))
```

Comment:
Separate issue for the tables, but I guess here we need to refer to the "contains everything" `Package` that you get from `pudl.metadata.classes.Package.from_resource_ids()` and implement a new method that's something like `get_resources_by_datasource()` which returns the IDs of all the tables that show up within a given source, based on their `etl_group` attributes.